### ISSUE TITLE
[6.0.100-preview.7.21362.10]Index was outside the bounds of the array when starting global tool app dotnet-format

### ISSUE THREAD

**Issue by Junjun-zhao** [dotnet/sdk#18876]
**Application Name**: dotnet-format (global tool app)
**OS:** Windows 10 RS5
**CPU**: X64
**.NET Build Number**: 6.0.100-preview.7.21362.10

**Verify Scenarios:**
1). Windows 10 RS5 X64 + .NET Core SDK build 6.0.100-preview.7.21362.10 Fail
2). Windows 10 RS5 X64 + .NET Core SDK build 6.0.100-preview.7.21360.1: Fail
3). Windows 10 RS5 X64 + .NET Core SDK build 6.0.100-preview.7.21356.4  Pass

**Repro steps:**
1) Install app with this command : dotnet tool install -g dotnet-format
2) On CMD, go to any project directory that has .sln or csproj file
3) Run this command : dotnet format

**Expected Result:**
It should format the files, no error

**Actual Result:**
```
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Microsoft.DotNet.CommandFactory.DotnetToolsCommandResolver.Resolve(CommandResolverArguments arguments) in dotnet.dll:token 0x60007aa+0x0
   at Microsoft.DotNet.CommandFactory.CompositeCommandResolver.Resolve(CommandResolverArguments commandResolverArguments) in dotnet.dll:token 0x6000796+0x15
   at Microsoft.DotNet.CommandFactory.CommandResolver.TryResolveCommandSpec(ICommandResolverPolicy commandResolverPolicy, String commandName, IEnumerable`1 args, NuGetFramework framework, String configuration, String outputPath, String applicationName) in dotnet.dll:token 0x6000826+0x45
   at Microsoft.DotNet.CommandFactory.CommandFactoryUsingResolver.Create(ICommandResolverPolicy commandResolverPolicy, String commandName, IEnumerable`1 args, NuGetFramework framework, String configuration, String outputPath, String applicationName) in dotnet.dll:token 0x6000774+0x0
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, TimeSpan startupTime, ITelemetry telemetryClient) in dotnet.dll:token 0x60009ae+0x311
   at Microsoft.DotNet.Cli.Program.Main(String[] args) in dotnet.dll:token 0x60009ac+0x6f
```
**Findings :**

"dotnet-format" command works as expected, "dotnet format" worked on previous SDKs, in this SDK, there is error.

@dotnet-actwx-bot @dotnet/compat

**Comment by sfoslund** [comment]
@JoeRobich is it enough for us to put the format tag on format issues or do you want to be tagged?

**Comment by JoeRobich** [comment]
@Junjun-zhao We are in the process of adding the format command to the CLI itself. It appears that we have broken the use of `dotnet format` at the moment. You should still be able to run the global tool by invoking as `dotnet-format`.

**Comment by jmarolf** [comment]
@sfoslund this is fixed by https://github.com/dotnet/sdk/pull/18840 which is currently in main. Do I need to port that to a specific branch for it to land in preview 7?

**Comment by sfoslund** [comment]
Nope, main is still preview 7 for now, thanks!

**Comment by Junjun-zhao** [comment]
Verified the issue and it has fixed on Preview 7 build dotnet-sdk-6.0.100-preview.7.21371.10 . so closing it.
